Datahub.Web/Startup.cs

```diff
@@ -44,13 +42,16 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddTransient<IElasticsearchService, ElasticsearchService>();
             services.AddTransient<ISearchBuilder, SearchBuilder>(); 
+            services.AddTransient<IS3Service, S3Service>();
```

Comment:
We are using an instance role to manage access, the key the box starts will be replaced periodically under the hood, so we need to be transient so the access keys are redone as I understood it, transient calls remade the service on injection.

Reply:
Yep "transient" dependencies (in the lingo of the asp net core dependency injection) are
> created each time they're requested from the service container. This lifetime works best for lightweight, stateless services. 

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-2.2#service-lifetimes

It says here that AWS client objects are not horribly expensive, but also that they are thread safe and designed to be reused, which is what I assumed (although it took me a while to find a link that confirmed!). 

https://aws.amazon.com/blogs/developer/object-lifecycles/

I'd have naturally made this a singleton lifetime in the AppDomain or at least request-scoped. But I think this can be left as-is tho since the link says that they're not too expensive.